Core/Object Arts/Dolphin/IDE/Professional/SystemBrowserShell.cls

```diff
@@ -135,7 +135,22 @@ selectionEnvironment
 		ifFalse: [classEnvironment]!
 
 slideyPinNames
-	^super slideyPinNames , #('packagesSlidey')! !
+	^super slideyPinNames , #('packagesSlidey')!
+
+updateCategories
+	"Overridden to select the appropriate loose category when viewing a non-owned class"
+
+	| class package |
+
+	class := self actualClass.
+	self hasPackageSelected ifTrue: [package := self packages anyOne].
+	(class notNil and: [package notNil and: [package includesLooseMethodsFor: class]]) ifFalse: [^super updateCategories].
+
+	categoriesPresenter model list: self actualClassChainCategories asOrderedCollection.
+	categoriesPresenter expand: self allCategory.
+	(LooseMethodCategory forPackage: package) 
```

Comment:
It seems undesirable architecturally to have a dependency on a particular type of virtual category in the browser. Although I think the outcome of displaying just the loose methods when a class with loose methods in the package is selected can be useful, I'm not convinced it should be the default, and I don't like the fact that changing class selection automatically makes a selection in the category tree.  I think it might be better to achieve similar functionality by just adding an option to show only owned methods of classes (which could be toggled on and off like the "inherited methods" filter), especially as this could also work for multiple package selections.